pkg/cache/cache.go

```diff
@@ -184,14 +203,32 @@ func (c *ClusterQueue) updateWorkloadUsage(wi *workload.Info, m int64) {
 	}
 }
 
+func (c *Cache) AddOrUpdateResourceFlavor(rf *kueue.ResourceFlavor) {
+	c.Lock()
+	c.resourceFlavors[rf.Name] = rf
+	for _, cq := range c.clusterQueues {
+		// We call update on all ClusterQueues irrespective of which CQ actually use this flavor
+		// because it is not expensive to do so, and is not worth tracking which ClusterQueues use
```

Comment:
Debatable, because we create maps. I would be slightly worried about restarts, as now the startup is `O(n*m)`, for number of CQs and RFs.

But for the scale we expect for n and m, it might be fine for now or even long term.